### PR TITLE
doc: add doc about dslx

### PR DESCRIPTION
**Pull request by woodgear**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ Now you visit the app by `curl http://${ip}`
 ### `Frontend` and `Rule`
 
 Complex traffic matching and distribution patterns can be configured by `Frontend` and `Rule`.
+[syntax of rule's dslx](./docs/rules.md)
```
Comment (oilbeater) [inline README.md:114]: Need an empty line between sentences

File: README.md [hunk 2]
```diff
@@ -216,10 +247,10 @@ spec:
 
 ### Gateway API
 
-ALB supports Gateway out of box, just set the `gatewayClassName` to `exclusive-gateway` when creating gateways.
+ALB supports GatewayApi(v0.6.2) out of box, just set the `gatewayClassName` to `exclusive-gateway` when creating gateways.
```
Comment (oilbeater) [inline README.md:250]:
```suggestion
ALB supports GatewayAPI(v0.6.2) out of box, just set the `gatewayClassName` to `exclusive-gateway` when creating gateways. GatewayAPI 1.0 support is on the way.
```

File: README.md [hunk 3]
```diff
@@ -137,11 +138,41 @@ metadata:
 spec:
   backendProtocol: ""                       # as same as Frontend
   certificate_name: ""                      # as same as Frontend
-  dslx:                                     # matching rule DSL
+  dslx:                                     # this rule match url start with /app-a or /app-b and method is post,and url param's group is vip, and host is *.app.com, and header's location is east-1 or east-2 and has a cookie name is uid,and source ip come from 1.1.1.1-1.1.1.100
```
Comment (oilbeater) [inline README.md:141]:
```suggestion
  dslx:                                     # this rule matches url starts with /app-a or /app-b and method is post,and url param's group is vip, and host is *.app.com, and header's location is east-1 or east-2 and has a cookie name is uid, and source IPs come from 1.1.1.1-1.1.1.100
```

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (woodgear) [reply README.md:114]: ok

**Review by oilbeater — APPROVED** [review]

**Pull request merged**